cmd/intervalContacts/intervalContacts.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/vertgenlab/gonomics/bed"
+	"github.com/vertgenlab/gonomics/bed/bedpe"
+	"github.com/vertgenlab/gonomics/exception"
+	"github.com/vertgenlab/gonomics/fileio"
+	"github.com/vertgenlab/gonomics/interval"
+	"log"
+)
+
+func intervalContacts(bedpeFile string, inFile string, contactOutFile string) {
+	var inIntervals = make([]interval.Interval, 0)
+	var currOverlaps []interval.Interval
+	var err error
+
+	out := fileio.EasyCreate(contactOutFile)
+
+	contactRecords := bedpe.Read(bedpeFile)
+	inChan := interval.GoReadToChan(inFile)
+
+	for i := range inChan {
+		inIntervals = append(inIntervals, i)
+	}
+	inTree := interval.BuildTree(inIntervals)
+
+	for _, i := range contactRecords {
+		currOverlaps = interval.Query(inTree, i.A, "any")
```

Comment:
I was reading interval.Query but I couldn't figure out if all corner cases would be accounted for here. For example, bed is [closed, open) so the last base is not part of the entry and inFile intervals that only overlap the end of a bed should not count. This particular case might not matter so much though since it's only 1 base. Do you think interval.Query has been giving you results that look real and reliable?

Reply:
In relation to this, I wonder if it will be helpful to include another field in the output file indicating "how much the inFile interval overlaps with the bedpe entry", whether it is 1 base out of 100, or 50 bases out of 100, etc.